### DISCUSSION TITLE
Methods to Determine Data Types (Numeric or Categorical) in PyCaret 3.0

### DISCUSSION THREAD

**Discussion by biniron** [pycaret/pycaret#3818] [Q&A]
Is there a method to determine if data types are classified as numeric or categorical in PyCaret 3.0?
In case this functionality has been discontinued in version 3.0, could you instruct on whether it is possible to alter the classification of data into numeric or categorical by adjusting data types in pandas?

**Comment by celestinoxp** [comment]
I don't know if I understand your question... normally when importing from Pycaret, the target data type is already distinguished

- numeric data: `from pycaret.regression import *`
- categorical data: `from pycaret.classification import *`

**Comment by tonpimenta** [comment]
I'm also having an issue with data types. I know you can specify the types of each feature at the `setup()` function, see here: [https://pycaret.gitbook.io/docs/get-started/preprocessing/data-preparation#data-types](url)
But in my case, it's not working, and I don't know why yet.

**Reply by celestinoxp** [reply]
It is not possible to analyze problems without having the code and the dataset, you must also share the error message!
Most of the time the error is the user's...
If you cannot share the code and dataset, create a reproducible example so that you can analyze the error...
you should also share which version of pycaret you are using and the version of python

**Reply by tonpimenta** [reply]
I've reported my issue here:
[https://github.com/pycaret/pycaret/discussions/3837](url)

@celestinoxp , if you please could take a look and see if you're able to assist me, I appreciate a lot! thanks

**Reply by celestinoxp** [reply]
I answered your problem..
note: I'm just a contributor and not part of the pycaret core team.
